mime.go

```diff
@@ -29,15 +29,22 @@ const (
 	yuyvMagicString    = "YUYV"
 	yuyvHeaderDimBytes = 4
 	yuyvBytesPerPixel  = 2
+	rgbaMagicString    = "RGBA"
+	rgbaHeaderDimBytes = 8
+	rgbaBytesPerPixel  = 4
 )
 
 type mimeHandler struct {
 	logger     logging.Logger
 	jpegEnc    *C.AVCodecContext
 	yuyvFrame  *C.AVFrame
 	yuyvSwsCtx *C.struct_SwsContext
+	rgbaFrame  *C.AVFrame
+	rgbaSwsCtx *C.struct_SwsContext
 	currentPTS int
-	mu         sync.Mutex
+	jpegMu     sync.Mutex
+	yuyvMu     sync.Mutex
+	rgbaMu     sync.Mutex
```

Comment:
Why the separate mutexes? Are there races we introduce and do we pass the mimehandler in by value anywhere?
https://stackoverflow.com/questions/54681678/multiple-mutexes-in-same-struct

I prefer having one mutex per struct to keep the read/write flow and locking behaviour explicit, especially since all of these are private, so we could separate them out or make nested structures.

Reply:
Each one protects each frame format's respective encoder/sws context. If we only used one, it would make concurrent different frame format GetImage requests slower since we're over-protecting.

Reply:
I think the code handles the mimehandler as a pointer everywhere

Reply:
So if we've got three clients - a vision service, data collector and the frontend all asking for different mimetypes at the same time, what happens with the three mime mutexes?

Reply:
IIUC this code:

The three mime mutexes are all utilized in that case. They wouldn't protect contention though because each is requesting a different mime type and there is only one subscriber per mime type.

The "intended" use of having per-mime mutexes would be if 2/3 of the subscribers all wanted one frame mimetype e.g. jpeg, but then the other 1/3 wanted a different one e.g. yuyv, then we would be properly managing contention because the yuyv can convert in parallel with the two jpeg subs do their lock contention thing.

If we had a single mutex, the yuyv would have to wait for the jpeg guys to not be currently converting, when it should just be able to do it freely.

Reply:
Talked IRL, limiting to 1 mutex for sanity and scope